src/components/Query.ts

```diff
@@ -51,25 +54,22 @@ export class Engine {
   ): Promise<INode[]> {
     this.outputChannel.show();
     this.outputChannel.appendLine(`${now()} - ${msg}`);
-    let result: Result;
-    if (conn.version !== InfluxConnectionVersion.V1) {
-      result = await APIRequest.Query(conn, query);
-    } else {
-      result = await APIRequest.QueryV1(conn, query, pp);
-    }
 
-    if (result.Err !== undefined) {
-      this.outputChannel.appendLine(`${now()} - Err: ${result.Err}`);
-    } else if (!result.Result) {
-      return [];
-    } else {
-      var nodes: Array<INode> = [];
-      for (let row of result.Result.Rows) {
-        nodes.push(newNodeFn(row[0], this.outputChannel, conn, pp));
+    let result: TableResult;
+    try {
+      if (conn.version === InfluxConnectionVersion.V1) {
```

Comment:
I really like the "tell don't ask" principle here; ideally we should be able to just call a single `query` function that'll work correctly based off of the connection version, instead of needing to ask the connection for its version first. Will take some more refactoring, so not necessary for this PR.